src/test/java/com/dropwizard/example/h2/resources/PersonResourceTest.java

```diff
@@ -0,0 +1,150 @@
+package com.dropwizard.example.h2.resources;
+
+import com.dropwizard.example.h2.DAO.PersonDAO;
+import com.dropwizard.example.h2.model.Person;
+import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
+import io.dropwizard.testing.junit5.ResourceExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class PersonResourceTest {
+
+    private static final PersonDAO personDAO = mock(PersonDAO.class);
+
+    public static final ResourceExtension RULE = ResourceExtension.builder()
+            .addResource(new PersonResource(personDAO))
+            .build();
+
+    private Person person1;
+    private Person person2;
+
+    @BeforeEach
+    void setup() {
+        person1 = new Person();
+        person1.setId(1L);
+        person1.setFirstName("firstname1");
+        person1.setLastName("lastname1");
+
+        person2 = new Person();
+        person2.setId(2L);
+        person2.setFirstName("firstname2");
+        person2.setLastName("lastname2");
+
+    }
+
+    @AfterEach
+    void tearDown() {
+        reset(personDAO);
+    }
+
+    @Test
+    public void getAllWithTwoElements() {
+        List<Person> persons = List.of(person1, person2);
+        when(personDAO.getAll()).thenReturn(persons);
+
+        List<Person> result = RULE.target("/persons").request().get(new GenericType<List<Person>>() {});
+
+        assertEquals(2, result.size());
+        assertEquals("firstname1", result.get(0).getFirstName());
+        assertEquals("firstname2", result.get(1).getFirstName());
+
+    }
+
+    @Test
+    public void getAllWithoutElements() {
+        List<Person> persons = List.of();
+        when(personDAO.getAll()).thenReturn(persons);
+
+        List<Person> result = RULE.target("/persons").request().get(new GenericType<List<Person>>() {});
+        assertEquals(0, result.size());
+        verify(personDAO, times(1)).getAll();
+    }
+
+
+    @Test
+    void getPersonSuccess() {
```

Comment:
About naming. If you test method _getPersonById_ here then name test method as _testGetPersonById_. If you want to specify result of invocation you can have it like _testGetPersonById_Success_. The rule is -> test<MethodName>_<Result or Very useful info to understand test>